editor/contexts/ThemeProvider.tsx

```diff
@@ -0,0 +1,40 @@
+"use client";
+import { useEffect, useState, createContext, useContext } from "react";
+import { darkModeLocalStorageKey } from "@/components/DarkModeToggle";
+
+export const ThemeContext = createContext<string | null>(null);
+
+/**
+ * Global context provider that returns whether the app is in dark or light mode
+ */
+export const ThemeProvider: React.FC<{ children: React.ReactNode }> = ({
```

Comment:
@johnclary this is the new `ThemeProvider`, this is my first time working with react global contexts and providers so lots of TILs for me today, does this all look right to you?

i do find it a little confusing that there is a React import called ThemeProvider as well, so could name our custom one something slightly diff to avoid confusion but i didnt have better ideas :shrug:

Reply:
in case it is helpful, this is a similar pattern that we use in Moped to make user/session details available throughout the app ([code](https://github.com/cityofaustin/atd-moped/blob/main/moped-editor/src/auth/user.js)). We also instantiated state from local storage, and it performed solidly.

we memoize there, but I don't see a need here since the state in this context contains a boolean.

anyways, this is really cool code to observe the theme changing!! 🚀

Reply:
This looks great!

It looks like it's `react-bootstrap` that exports a `ThemeProvider` as well. That's annoying, but I think what you have is fine. Alternatively: `AppThemeProvider`?